### ISSUE TITLE
Build fails with libmicrohttpd-0.9.71

### ISSUE THREAD

**Issue by kripton** [OpenLightingProject/ola#1650]
libmicrohttpd-0.9.71 changed the return type of some of its functions and as such the build of ola-0.10.* now fails.
References:
* https://git.gnunet.org/libmicrohttpd.git/commit/?id=6347f514aa2388e774d5bf356df8046864e5f73c
* https://github.com/xbmc/xbmc/pull/18131
* https://bugs.archlinux.org/task/67208

**Comment by kripton** [comment]
Error log:
```
common/http/HTTPServer.cpp: In member function 'bool ola::http::HTTPRequest::Init()':
common/http/HTTPServer.cpp:221:60: error: invalid conversion from 'int (*)(void*, MHD_ValueKind, const char*, const char*)' to 'MHD_KeyValueIterator' {aka 'MHD_Result (*)(void*, MHD_ValueKind, const char*, const char*)'} [-fpermissive]
  221 |   MHD_get_connection_values(m_connection, MHD_HEADER_KIND, AddHeaders, this);
      |                                                            ^~~~~~~~~~
      |                                                            |
      |                                                            int (*)(void*, MHD_ValueKind, const char*, const char*)
In file included from ./include/ola/http/HTTPServer.h:42,
                 from common/http/HTTPServer.cpp:29:
/usr/include/microhttpd.h:2699:49: note:   initializing argument 3 of 'int MHD_get_connection_values(MHD_Connection*, MHD_ValueKind, MHD_KeyValueIterator, void*)'
 2699 |                            MHD_KeyValueIterator iterator,
```

**Comment by kripton** [comment]
@peternewman : Would you prefer one `ifdef` per affected function all OR a new ´define` to use as the data type as done in https://git.gnunet.org/gnunet.git/tree/src/include/gnunet_mhd_compat.h ?

**Comment by peternewman** [comment]
Oh, presumably `--disable-fatal-warnings` allows the build to succeed for now?

**Comment by peternewman** [comment]
> @peternewman : Would you prefer one `ifdef` per affected function all OR a new `define` to use as the data type as done in https://git.gnunet.org/gnunet.git/tree/src/include/gnunet_mhd_compat.h ?

I'm torn. We'd normally do the former, but given it's in the return type or generally in the calling parameters it makes it rather messy, especially because of the way it's used as a callback. I suspect what they've proposed in gnunet is the least confusing option.

**Comment by kripton** [comment]
> Oh, presumably `--disable-fatal-warnings` allows the build to succeed for now?

Yes, I assume that making warnings non-fatal might make the build succeed. However, if #1651 is good, that should not be necessary.

**Comment by kripton** [comment]
Fixed by #1651 => closing :)
